src/crimson/osd/watch.cc

```diff
@@ -169,10 +182,38 @@ seastar::future<> Notify::maybe_send_completion()
       CEPH_WATCH_EVENT_NOTIFY_COMPLETE,
       empty,
       client_gid);
-    reply->set_data(bl);
+    ceph::bufferlist reply_bl;
+    {
+      std::list<std::pair<uint64_t,uint64_t>> missed;
```

Comment:
nit, could use a `vector<std::pair<>>`. i know, classic osd uses a `list<>`, but i think we should not follow it. probably better off having another PR fix the classic osd as well.

Reply:
the change on the classic osd end is created at #39882

Reply:
Sure!